boa/src/builtins/iterable/mod.rs

```diff
@@ -203,20 +203,6 @@ impl IteratorRecord {
 
 #[derive(Debug)]
 pub struct IteratorResult {
-    value: JsValue,
-    done: bool,
-}
-
-impl IteratorResult {
-    fn new(value: JsValue, done: bool) -> Self {
-        Self { value, done }
-    }
-
-    pub fn is_done(&self) -> bool {
-        self.done
-    }
-
-    pub fn value(self) -> JsValue {
-        self.value
-    }
+    pub value: JsValue,
+    pub done: bool,
```

Comment:
This I don't like. Is this needed? Using public members makes this structure modifiable almost anywhere and with no control, and makes it much more difficult if we ever want to change the inner structure.

The getter and setter patterns are very well established in Rust for a reason. They are less error prone and easier to maintain external compatibility.

Reply:
Using getters and setters is justifiable if we had to maintain some state internally. This struct is just a wrapper for a `(Value, bool)` to represent an iterator item, I don't see the utility on having only getters if the struct will have just `is_done` and `get_value` as methods.

Reply:
While you are right that there is no functionality with this struct, I think the private fields preserve the nature of the struct. It should only be used to access the result of an iterator next call. If the fields are public, the struct is really only a wrapper for `(JsValue, bool)`. In that case we could also just work with that tuple directly instead of using a struct.

Reply:
We could return a tuple, but using an struct makes more explicit what each type represents. That's the same reason why it's more idiomatic to represent a 2d point as a `Point {x: f64, y: f64}` instead of a `(f64, f64)`.

Reply:
But, if it's not meant to be modified, we don't even need the setters, and not having them would prevent accidental mutation.

Reply:
But it is meant to be modified. Or at least is meant to be destructured and used as an owned value, like in:

https://github.com/boa-dev/boa/blob/2f8c35db571e3a1c4dd03040d70cafcc02d69ae9/boa/src/syntax/ast/node/array/mod.rs#L54-L60

or in 

https://github.com/boa-dev/boa/blob/2f8c35db571e3a1c4dd03040d70cafcc02d69ae9/boa/src/syntax/ast/node/new/mod.rs#L61-L66

And no function takes an `IteratorResult` as a parameter, so we don't need to ensure that `IteratorResult` is not modified. If we used `IteratorResult` as a parameter on any function, I would completely agree with using getters.

Reply:
Now that I think about this, is there a reason that `IteratorResult` is `pub` instead of `pub(crate)` at all? We neither return nor accept this anywhere in the public api.

Reply:
> Now that I think about this, is there a reason that `IteratorResult` is `pub` instead of `pub(crate)` at all? We neither return nor accept this anywhere in the public api.

Yeah, I just copied the current visibility, but maybe we shouldn't expose it to the user.

Reply:
The API is not very well defined yet, and we will probably need to change many visibilities. I'm OK with these changes.